### DISCUSSION TITLE
Recorded membrane potential values from `arbor.cable_probe_membrane_voltage_cell`

### DISCUSSION THREAD

**Discussion by espenhgn** [arbor-sim/arbor#2047] [Q&A] [answered]
Dear developers; 
I just have a question about the interpretation of recorded membrane potential values using the [`arbor.cable_probe_membrane_voltage_cell()`](https://docs.arbor-sim.org/en/latest/python/probe_sample.html#arbor.cable_probe_membrane_voltage_cell) API.

I'm pretty sure I've read *somewhere* that the recorded values correspond to the spatially average membrane potential within each CV, but now I cannot find that reference anymore. Is this the correct interpretation?

On a related note, is the interpolation operation performed by the [`arbor.cable_probe_membrane_voltage()`](https://docs.arbor-sim.org/en/latest/python/probe_sample.html#arbor.cable_probe_membrane_voltage) API a linear interpolation between values in the corresponding CVs?

**Comment by thorstenhater** [comment, marked as answer]
Hi @espenhgn 

it might have been the commented source code in `cable_cell.hpp` ll61
```cxx
// Voltage estimate [mV] at `location`, interpolated.
// Sample value type: `double`
// Sample metadata type: `mlocation`
struct ARB_SYMBOL_VISIBLE cable_probe_membrane_voltage {
    locset locations;
};

// Voltage estimate [mV], reported against each cable in each control volume. Not interpolated.
// Sample value type: `cable_sample_range`
// Sample metadata type: `mcable_list`
struct ARB_SYMBOL_VISIBLE cable_probe_membrane_voltage_cell {};
```
by that no interpolation is performed for the `_cell` probes, rather each (=spatial) value is reported as-is.

The term interpolation itself is a bit tricky, if I understand correctly there's two cases that can occur:
- trivial: we have hit a single CV, interpolation is a no-op and returns the voltage at the CV's center
- fork: However, if we hit a CV where the morphology is forking, we return a linear interpolation between the
  forking CV's parent and itself.

Does that help?

**Reply by espenhgn** [reply]
Hi. I now see in the C++ docs for `struct cable_probe_membrane_voltage_cell` (https://docs.arbor-sim.org/en/v0.8/cpp/probe_sample.html#membrane-voltage), 1st bullet: "Sample value: `cable_sample_range`. Each value is the **average** membrane potential in millivolts across an unbranched component of the cell, as determined by the discretisation".  

Thanks for the clarification on the interpolation question.
